src/DIRAC/Core/Utilities/DictCache.py

```diff
@@ -64,6 +65,8 @@ def __init__(self, deleteFunction=False, threadLocal=False):
         # Function to clean the elements
         self.__deleteFunction = deleteFunction
 
+        self.__finalizer = weakref.finalize(self, self.purgeAll, useLock=False)
```

Comment:
The use of `purgeAll` here should probably be refactored to not depend on self and go pass in the members it needs instead.

Reply:
Fine by me, do you want to add it in this PR ?

Reply:
Done